### PR TITLE
Small improvments on assumptions section

### PR DESCRIPTION
**Pull request by certik**
1. fix typo (double real assumption)
2. Clarify False return value, add example on this.

### CODE REVIEW

**Comment by certik** [comment]
@skirpichev wants to put in some more patches, so please do not merge until I get an OK from him.

**Comment by certik** [comment]
@asmeurer would you mind reviewing this?

**Comment by certik** [comment]
I went over the changes and they all look good to me, but if you could @asmeurer also go over it, that would be great.

Regarding authorship, Sergey has contributed to sympy (obviously) and also enough to this article with this PR, so by merging this he becomes a co-author, and that's fine with me.

**Comment by certik** [comment]
P.S. Sergey said to merge this PR if there are no objections, and that he will fix his patches if there are objections.

File: architecture.tex [hunk 1]
```diff
@@ -264,12 +268,12 @@ \subsection{Extensibility}
...
 
 \begin{lstlisting}[caption={A minimal implementation of \texttt{sympy.gamma}.},label=fig:gamma-example]
-from sympy import Integer, Function, floor, factorial, polygamma
+from sympy import Function, floor, factorial, polygamma
 
 class gamma(Function)
     @classmethod
     def eval(cls, arg):
-        if isinstance(arg, Integer) and arg.is_positive:
+        if arg.is_integer and arg.is_positive:
```
Comment (asmeurer) [inline architecture.tex]: I would leave this as it was. `gamma` in SymPy evaluates on Integer, not integer, and regardless, evaluating based on assumptions not something I particularly want to encourage.

File: introduction.tex [hunk 2]
```diff
@@ -37,7 +37,7 @@
 
 As a library, SymPy does not have a built-in graphical user
 interface (GUI). However, SymPy exposes a rich interactive display system,
-including registering printers with Jupyter~\cite{perez2007ipython} frontends,
+supports registering printers with Jupyter~\cite{perez2007ipython} frontends,
```
Comment (asmeurer) [inline introduction.tex]: "and supports"?

File: solvers.tex [hunk 3]
```diff
@@ -29,5 +29,5 @@
 
 \texttt{solveset} is under active development as a planned replacement for
 \texttt{solve}. There are certain features which are implemented in
-\texttt{solve} that are not yet implemented in \texttt{solveset}. Notably,
-these include nonlinear multivariate system and transcendental equations.
+\texttt{solve} that are not yet implemented in \texttt{solveset},
+these include multivariate systems and some transcendental equations.
```
Comment (asmeurer) [inline solvers.tex]: Comma splice

**Comment by asmeurer** [comment]
Some minor issues. If they aren't fixed by the time Travis builds I'll merge and fix them myself.

**Comment by certik** [comment]
Just fix it yourself, as I am going to bed now (Sergey sends patches to me, I push it in). He can send them to you as well, then you can push it in.

**Comment by certik** [comment]
This is +1 from me to go in.

@asmeurer feel free to merge this, if this is ok with you.

**Pull request merged**